### PR TITLE
DHFPROD-6085:Allow removal of properties from step when loading from gradle

### PR DESCRIPTION
**Pull request by srinathgit**
### Description
Allow removal of properties from step when loading from gradle
#### Checklist: 
```diff
- Note: do not change the below
```

-  ##### Owner:

- [x] JIRA_ID included in all the commit messages
- [x] PR title is in the format JIRA_ID:Title
- [x] Rebase the branch with upstream
- [x] Squashed all commits into a single commit
- [x] Added Tests
  

- ##### Reviewer:

- [x] Reviewed Tests

### CODE REVIEW

File: marklogic-data-hub/src/main/java/com/marklogic/hub/dataservices/StepService.java [hunk 1]
```diff
@@ -140,21 +147,30 @@ private void deleteStep(BaseProxy.DBFunctionRequest request, String stepDefiniti
...
     com.fasterxml.jackson.databind.JsonNode getStepsByType(Stream<String> propertiesToReturn);
 
+    /**
+     * Invokes the saveStep operation on the database server
+     *
+     * @param stepDefinitionType	provides input
+     * @param stepProperties	provides input
+     * @return	Return the created/updated step document
+     */
+    com.fasterxml.jackson.databind.JsonNode saveStep(String stepDefinitionType, com.fasterxml.jackson.databind.JsonNode stepProperties);
```
Comment (rjrudin) [inline marklogic-data-hub/src/main/java/com/marklogic/hub/dataservices/StepService.java:157]: Shouldn't DS have removed this method? I think we only want one saveStep method, and I think it's good that the client has to explicitly choose whether to do an overwrite or not.

Comment (rjrudin) [reply marklogic-data-hub/src/main/java/com/marklogic/hub/dataservices/StepService.java:157]: I just tried this locally - I added overwrite:boolean to the API, then ran "gradle generatestepinterface", and the method was updated to include the boolean. That's what we want, we don't want a saveStep method that doesn't require the boolean.

File: marklogic-data-hub/src/test/java/com/marklogic/hub/dataservices/customStep/CreateAndUpdateCustomStep.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.marklogic.hub.AbstractHubCoreTest;
+import com.marklogic.hub.dataservices.StepService;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+public class CreateAndUpdateCustomStep extends AbstractHubCoreTest {
```
Comment (rjrudin) [inline marklogic-data-hub/src/test/java/com/marklogic/hub/dataservices/customStep/CreateAndUpdateCustomStep.java:12]: Needs to have "Test" suffix so we know it's a test

File: marklogic-data-hub/src/test/java/com/marklogic/hub/dataservices/customStep/CreateAndUpdateCustomStep.java [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    @Test
+    void customStep() {
+        ObjectMapper mapper = new ObjectMapper();
```
Comment (rjrudin) [inline marklogic-data-hub/src/test/java/com/marklogic/hub/dataservices/customStep/CreateAndUpdateCustomStep.java:15]: Can use objectMapper in the parent class

**Review by rjrudin — CHANGES_REQUESTED** [review]
Looks good, just a couple tweaks

**Review by rjrudin — APPROVED** [review]
This looks good

**Review by akshaysonvane — APPROVED** [review]

**Pull request merged**